front_end/panels/react_devtools/ReactDevToolsView.ts

```diff
@@ -128,47 +126,29 @@ export class ReactDevToolsViewImpl extends UI.View.SimpleView {
 
   private sendMessage(event: string, payload?: ReactDevToolsTypes.MessagePayload): void {
     for (const model of SDK.TargetManager.TargetManager.instance().models(ReactDevToolsModel, {scoped: true})) {
-      ++this.messageQueueSize;
-
-      void model.sendMessage({event, payload}).finally(() => {
-        --this.messageQueueSize;
-        this.checkIfReloadShouldBeResumed();
-      });
+      void model.sendMessage({event, payload});
     }
   }
 
-  private checkIfReloadShouldBeResumed(): void {
-    if (this.messageQueueSize === 0 && this.pageWillBeReloaded) {
-      this.pageWillBeReloaded = false;
-
-      const resourceTreeModel = this.resourceTreeModel;
-      if (resourceTreeModel === null) {
-        throw new Error('Page.reload event should have been resumed, but ResourceTreeModel is null');
-      }
-
-      resourceTreeModel.resumeReload();
-      this.resourceTreeModel = null;
+  private onExecutionContextDestroyed(event: ContextDestroyedEvent): void {
+    if (event.data.name !== 'main') {
+      return;
     }
-  }
 
-  private onPageReloadRequested(event: PageReloadRequestedEvent): void {
     // Unmount React DevTools view
     this.contentElement.removeChildren();
 
-    const resourceTreeModel = event.data;
-    this.resourceTreeModel = resourceTreeModel;
-
-    // We will resume reload once all messages are sent, it is important to notify RDT backend
-    this.pageWillBeReloaded = true;
-    resourceTreeModel.suspendReload();
-
     this.bridge.shutdown();
     this.listeners.clear();
 
     this.renderLoader();
   }
 
-  private onExecutionContextCreated(): void {
+  private onExecutionContextCreated(event: ContextCreatedEvent): void {
+    if (event.data.name !== 'main') {
```

Comment:
1. Maybe include a comment with a link to https://github.com/facebook/react-native/blob/40b54ee671e593d125630391119b880aebc8393d/packages/react-native/ReactCommon/jsinspector-modern/InstanceTarget.cpp#L61 ?
2. Consider putting this in `const kMainExecutionName = 'main'` at the top of the file.

Reply:
Completely forgot about it, just added, thanks!